### DISCUSSION TITLE
How to adjust the maxFeePerGas parameter to a higher value when replaying a tx with era_test_node

### DISCUSSION THREAD

**Discussion by donoso-eth** [zkSync-Community-Hub/zksync-developers#284] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

I am replaying a successful [tx](https://explorer.zksync.io/tx/0xab2d71aefdc09c8708d2255e74f9d47c4135ac5af723943fabaa4cc66f65165f) on mainnet.

Trace logs for these transactions will really help us:
https://explorer.zksync.io/tx/0x1b7eb583504cf130aec8d180218d20d0ac9d11912ca70e5bf32624e221550ad1
https://explorer.zksync.io/tx/0x927a63a7b88507137d7db31a5d0d4a8f16a36ed0388810c76577914bd1b3884a
https://explorer.zksync.io/tx/0x5aad49b80779ea9300059a8f84c4f1bd0ffb19bb4957da94beb0963959f8d8c5

### What is the specific issue or error you're encountering?

Submitted Tx is Unexecutable 0xab2d71aefdc09c8708d2255e74f9d47c4135ac5af723943fabaa4cc66f65165f because of MaxFeePerGasTooLow 100000000

### Can you share the error messages or logs you're receiving, if any?

Submitted Tx is Unexecutable 0xab2d71aefdc09c8708d2255e74f9d47c4135ac5af723943fabaa4cc66f65165f because of MaxFeePerGasTooLow 100000000

### Have you made any recent changes to the contract before encountering this issue?

n/a

### Are there any external libraries or contracts that your contract interacts with?

n/a

### Can you provide the relevant portions of your contract code where the issue is occurring?

n/a

### Have you tried to isolate the problem, and if so, what were the results?

n/a

### What steps have you already taken to try to resolve the issue?

n/a

**Comment by dutterbutter** [comment]
@MexicanAce we may need to update `era-test-node` to latest zksync-era. It could be due to gas model changes if it succeeds on mainnet and not era-test-node. Thoughts? 

If need be while we investigate updating era-test-node or making params configurable during replay tx. You can do the following to adjust variables:

1. `git clone git@github.com:matter-labs/era-test-node.git` 
2. Update `MaxFeePerGas` here: https://github.com/matter-labs/era-test-node/blob/main/src/node/in_memory.rs#L1260 to something greater than `250000000` 
3. Run: `cargo build --release` 
4. Run test-node `./target/release/era_test_node --show-calls=all --resolve-hashes --show-gas-details=all replay_tx mainnet 0xab2d71aefdc09c8708d2255e74f9d47c4135ac5af723943fabaa4cc66f65165f ` 
5. Observe output

Hope that helps in the interim. We will look to add this feature / update. @donoso-eth

**Reply by donoso-eth** [reply]
@dutterbutter , thks I don't receive the error as before. However, it still doesn't work:
I've tested with different successful txs, and now the error for all txs is:
<img width="1383" alt="image" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/30406536/01f4658d-12c6-454c-bcc9-6991e7bc04fe">

Could it be that the tx replayed is not the one I passed, here two examples

1-.`/target/release/era_test_node --show-calls=all --resolve-hashes --show-gas-details=all replay_tx mainnet 0xe27f22b9fd17290414025c5d4b37b5dff50a179e0d80a5b041ebad3395f75157`
```
08:21:02  INFO Validating 0xa1a50c100ad3eb6ca22e306008383ea7ca464dc000ba9abb92ec890594016cf6
08:21:02  INFO Executing 0xa1a50c100ad3eb6ca22e306008383ea7ca464dc000ba9abb92ec890594016cf6
```
2- `./target/release/era_test_node --show-calls=all --resolve-hashes --show-gas-details=all replay_tx mainnet 0x1b7eb583504cf130aec8d180218d20d0ac9d11912ca70e5bf32624e221550ad1`
```
08:20:31  INFO Validating 0x8738c873bb41bda23b7524b55978341bfda2f4dfad2276171ee75a2cca207695
08:20:31  INFO Executing 0x8738c873bb41bda23b7524b55978341bfda2f4dfad2276171ee75a2cca207695
```

The complete logs here:
[Terminal Saved Output.txt](https://github.com/zkSync-Community-Hub/zksync-developers/files/14021000/Terminal.Saved.Output.txt)

**Reply by MexicanAce** [reply]
@dutterbutter yes, looks like `era-test-node` needs to be updated. In the meantime, @donoso-eth perhaps you can use the `debug_traceTransaction` RPC endpoints to get the trace you're looking for?

Docs: https://docs.zksync.io/build/api.html#debug-tracetransaction

**Reply by MexicanAce** [reply, marked as answer]
With the latest release, `era_test_node replay_tx` should be working again. We've tracked the additional request to pass in `maxFeePerGas` in that project's backlog

**Comment by tus4ncockverb** [comment]
The issue is with the submitted transaction being unexecutable due to the MaxFeePerGasTooLow error, specifically with a value of 100000000. To resolve this, you may need to adjust the MaxFeePerGas parameter to a higher value to meet the gas fee requirements for executing the transaction successfully on the mainnet.

**Comment by longbipubacr** [comment]
The submitted transaction is encountering the error "Unexecutable 0xab2d71aefdc09c8708d2255e74f9d47c4135ac5af723943fabaa4cc66f65165f because of MaxFeePerGasTooLow 100000000." This error indicates that the maximum fee per gas specified in the transaction is too low, resulting in the transaction being unexecutable. To resolve this issue, you need to increase the maximum fee per gas to a value that allows the transaction to be executed successfully.

**Comment by heuresfanegarnold** [comment]
The submitted transaction is failing due to the MaxFeePerGasTooLow error, indicating that the maximum fee per gas is too low for the transaction to be executed successfully. To resolve this issue, you may need to adjust the maximum fee per gas value to a higher value that meets the requirements for executing the transaction on the mainnet. Reviewing the transaction trace logs provided in the links may help in understanding the specific gas fee requirements for the transaction.

**Comment by yamyv** [comment]
Issue: Submitted transaction is unexecutable due to MaxFeePerGasTooLow.

Solution: Increase the MaxFeePerGas value to resolve the issue. You may need to adjust the gas fee parameters to ensure the transaction can be executed successfully.

**Comment by shipefred** [comment]
Environment: Mainnet
Functionality: Replaying successful transactions on mainnet.
Issue: Submitted transaction is unexecutable due to MaxFeePerGasTooLow error.
Error Message: Submitted Tx is Unexecutable 0xab2d71aefdc09c8708d2255e74f9d47c4135ac5af723943fabaa4cc66f65165f because of MaxFeePerGasTooLow 100000000

**Comment by wadejackie** [comment]
The user is attempting to replay successful transactions on the mainnet but is encountering an error message indicating that the submitted transaction is unexecutable due to MaxFeePerGasTooLow. The user has provided links to the relevant transactions but has not provided any additional details or repository link. Further investigation may be needed to determine the cause of the error and potential solutions.
